lib/luacrypto_aead.c

```diff
@@ -0,0 +1,258 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for AEAD
+* (Authenticated Encryption with Associated Data) ciphers.
+*
+* This module provides a `new` function to create AEAD transform objects,
+* which can then be used for encryption and decryption.
+*
+* @module crypto_aead
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/aead.h>
+#include <linux/err.h>
+#include <linux/scatterlist.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+#include "luacrypto.h"
+
+LUNATIK_PRIVATECHECKER(luacrypto_aead_check, struct crypto_aead *);
+
+LUACRYPTO_RELEASE(aead, struct crypto_aead, crypto_free_aead, );
```

Comment:
do we need this comma at the end? it looks odd to me.. if we need, probably it's better to use vararg.. there's some examples on lunatik.h